### PR TITLE
blog: This month in Servo (November 2024)

### PR DESCRIPTION
**Pull request by delan**

### CODE REVIEW

File: _posts/2024-12-05-this-month-in-servo.md [hunk 1]
```diff
@@ -1,40 +1,52 @@
...
 layout:     post
 tags:       blog
-title:      "This month in Servo: and more!"
+title:      "This month in Servo: grid layout, parallel flexbox, smaller binaries, and more!"
 date:       2024-12-05
 summary:    ao!! wrrrrao!!
 categories:
 ---
 
+Servo now supports **CSS grid layout** (@nicoburns, @taniishkaa, #32619, #34352, #34421), **:is() and :where() selectors** (@mrobinson, #34066), and the **getClientRects()** method on Element (@chickenleaf, #34025)!
```
Comment (jdm) [inline _posts/2024-12-05-this-month-in-servo.md]: getClientRects already existed. We just changed the return type to match some specification updates.

File: _posts/2024-12-05-this-month-in-servo.md [hunk 2]
```diff
@@ -19,6 +19,20 @@ Last month we also landed support for **‘justify-self’** on positioned eleme
...
 
+## More engine changes
+
+Layout performance continues to improve, with faster layout for **‘column’ flex containers** (@Loirooriol, @mrobinson, #34346) and faster caching of intrinsic sizes (@Loirooriol, #34384), in addition to parallel layout for flexbox (@mrobinson, #34132).
+
+We’ve fixed the behaviour of **offsetLeft** and **offsetTop** relative to &lt;body> with ‘position: static’ (@nicoburns, @Loirooriol, #32761), which also required spec changes (@nicoburns, @Loirooriol, csswg-drafts#10549).
+We’ve also fixed several layout bugs around:
+
+- **‘object-fit’** when ‘aspect-ratio’ is set (@Loirooriol, #34413)
+- **clientWidth** and friends on floats and replaced inlines ()
```
Comment (jdm) [inline _posts/2024-12-05-this-month-in-servo.md:30]: Missing author and PR.

Comment (delan) [reply _posts/2024-12-05-this-month-in-servo.md]: fixed in 6028773e154c465255507917fc753fa623a980ec

File: _posts/2024-12-05-this-month-in-servo.md [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+We’ve also added support for two key Shadow DOM interfaces, the **shadowRoot property** on Element (@simonwuelker, #34306) and the **innerHTML property on ShadowRoot** (@simonwuelker, #34335).
+
+Last month we also landed support for **‘justify-self’** on positioned elements (@chickenleaf, #34235), form submission with **&lt;input type=image>** (@shanehandley, #34203), **DataTransfer** (@Gae24, #34205), the **close()** method on ImageBitmap (@simonwuelker, #34124), plus several new **[SubtleCrypto](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto) API** features:
+
+- crypto.subtle.**deriveKey()** (@simonwuelker, #34185)
+- **wrapKey()** and **unwrapKey()** (@simonwuelker, #34262)
+- **“HMAC”** algorithm in **generateKey()** (#34278), **sign()**, **verify()** (#34223), and **deriveKey(`"get key length"`)** (#34230)
+- **“AES-GCM”** algorithm in **encrypt()**, **decrypt()**, **generateKey()**, and **importKey()** (#34269)
+- **“AES-KW”** algorithm in **generateKey()**, **importKey()**, **wrapKey()**, and **unwrapKey()** (#34262)
```
[the remark this answers is not in the document]
Comment (delan) [reply _posts/2024-12-05-this-month-in-servo.md]: fixed in 6028773e154c465255507917fc753fa623a980ec

File: _posts/2024-12-05-this-month-in-servo.md (hunk 1 above)
Comment (delan) [reply _posts/2024-12-05-this-month-in-servo.md]: fixed in c2e66f694653e490261e440ec2a24c249bf17b3c

File: _posts/2024-12-05-this-month-in-servo.md (hunk 2 above)
Comment (delan) [reply _posts/2024-12-05-this-month-in-servo.md]: fixed in dd61940d3b8133a1422522bea717e8612f5228b7

File: _posts/2024-12-05-this-month-in-servo.md [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+categories:
+---
+
+Servo now supports **CSS grid layout** (@nicoburns, @taniishkaa, #32619, #34352, #34421), **parallel layout for flexbox** (@mrobinson, #34132), **:is() and :where() selectors** (@mrobinson, #34066), and the **getClientRects()** method on Element (@chickenleaf, #34025)!
+We’ve also added support for two key Shadow DOM interfaces, the **shadowRoot property** on Element (@simonwuelker, #34306) and the **innerHTML property on ShadowRoot** (@simonwuelker, #34335).
+
+Last month we also landed support for **‘justify-self’** on positioned elements (@chickenleaf, #34235), form submission with **&lt;input type=image>** (@shanehandley, #34203), **DataTransfer** (@Gae24, #34205), the **close()** method on ImageBitmap (@simonwuelker, #34124), plus several new **[SubtleCrypto](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto) API** features:
+
+- crypto.subtle.**deriveKey()** (@simonwuelker, #34185)
+- **wrapKey()** and **unwrapKey()** (@simonwuelker, #34262)
```
[the remark this answers is not in the document]
Comment (delan) [reply _posts/2024-12-05-this-month-in-servo.md]: fixed in dd61940d3b8133a1422522bea717e8612f5228b7

File: _posts/2024-12-05-this-month-in-servo.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (delan) [reply _posts/2024-12-05-this-month-in-servo.md]: fixed in dd61940d3b8133a1422522bea717e8612f5228b7

File: _posts/2024-12-05-this-month-in-servo.md (hunk 1 above)
Comment (delan) [reply _posts/2024-12-05-this-month-in-servo.md]: fixed in efc40787fa225cafa8e32fd09b621dd460be0ca7

**Comment by sagudev** [comment]
BTW It might be worth noting that we archived sparkle, because we do not use it anymore (less maintenance burden).

File: _posts/2024-12-09-this-month-in-servo.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+title:      "This month in Servo: :is() and :where(), grid layout, parallel flexbox, and more!"
+date:       2024-12-09
+summary:    Plus shadow DOM buffs, smaller binaries, and new SubtleCrypto algorithms (HMAC, AES-GCM, AES-KW, PBKDF2).
+categories:
+---
+
+<figure class="_figr"><a href="{{ '/img/blog/november-2024.png' | url }}"><img src="{{ '/img/blog/november-2024.png' | url }}" style="width: 33em;"
+    alt="Servo nightly showing new support for CSS grid layout"></a></figure>
+
+Servo now supports **CSS grid layout** (@nicoburns, @taniishkaa, #32619, #34352, #34421), **parallel layout for flexbox** (@mrobinson, #34132), and **:is() and :where() selectors** (@mrobinson, #34066)!
```
Comment (mrego) [inline _posts/2024-12-09-this-month-in-servo.md:13]: Should we mention grid layout is not enabled by default and that you need to add a flag.

Comment (delan) [reply _posts/2024-12-09-this-month-in-servo.md]: done in a4e23e62454a8762be1bb79a0b550d06a2d2d6b5

Comment (delan) [reply _posts/2024-12-09-this-month-in-servo.md]: done in 15436f368661f0c58f65c7e86e16dff6e9cb1267

**Pull request merged**